src/cabinUt/cabinUt.service.ts

```diff
@@ -1,76 +1,43 @@
 import { Injectable } from '@nestjs/common';
-import { VisbookService } from 'src/visbook/visbook.service';
-import { dateIsValid, getVisbookId, isReservationsUrl } from './cabinUt.utils';
+import { SupabaseCabin } from 'src/cabinDatabase/cabinDatabase.interface';
 import { CabinUtApi } from './cabinUt.api';
-import { CabinUtDetails } from './cabinUt.interface';
+import { getVisbookId } from './cabinUt.utils';
 
 @Injectable()
 export class CabinUtService {
-  private readonly apiClient = new CabinUtApi();
+  constructor(private readonly cabinUtApi: CabinUtApi) {}
 
-  constructor(private readonly visbookService: VisbookService) {}
-
-  async getRandomCabin(): Promise<CabinUtDetails> {
-    const cabins = await this.apiClient.getCabins();
-    const cabin = cabins[Math.floor(Math.random() * cabins.length)];
-    const { id } = cabin.node;
-    const cabinDetails = await this.apiClient.getCabinDetails(id);
-    return cabinDetails;
+  async getCabins() {
+    return this.cabinUtApi.getCabins();
   }
 
-  async isCabinAvailable(
-    cabin: CabinUtDetails,
-    checkIn: string,
-    checkOut: string,
-  ): Promise<boolean> {
-    if (!dateIsValid(checkIn) || !dateIsValid(checkOut)) {
-      console.log('invalid date(s)');
-      return false;
-    }
-
-    const cabinHasReservationsUrl = isReservationsUrl(cabin.bookingUrl);
-    const cabinVisbookId = getVisbookId(cabin.bookingUrl);
-
-    if (!cabinHasReservationsUrl && cabinVisbookId === 0) {
-      return false;
-    }
-
-    const cabinIsAvailable = await this.visbookService.isCabinAvailable(
-      cabinVisbookId,
-      checkIn,
-      checkOut,
-    );
-
-    return cabinIsAvailable;
-  }
-
-  async getRandomAvailableCabin(
-    checkIn: string,
-    checkOut: string,
-  ): Promise<CabinUtDetails | undefined> {
-    if (!dateIsValid(checkIn) || !dateIsValid(checkOut)) {
-      return undefined;
-    }
-
-    const hasExceededTimeLimit = (now: number, timeLimit: number): boolean => {
-      const elapsedTime = now - startTime;
-      if (elapsedTime >= timeLimit) {
-        return true;
-      }
-      return false;
-    };
-
-    const startTime = Date.now();
-    let cabin = await this.getRandomCabin();
-
-    while ((await this.isCabinAvailable(cabin, checkIn, checkOut)) === false) {
-      cabin = await this.getRandomCabin();
-      const ONE_MIN_IN_MILLISECONDS = 1 * 60 * 1000;
-      if (hasExceededTimeLimit(Date.now(), ONE_MIN_IN_MILLISECONDS)) {
-        return undefined;
-      }
+  async getCabinDetailsAsSupabaseCabins(
```

Comment:
I think it would be better if this service didn't know about supabase. Means we could change DB services without needing to touch this module at all. Perhaps we can return an UtCabin?

Reply:
Actually I think this is clumsy  naming from my side. A SupabaseCabin is for the most part just a summary of the CabinDetails plus extracted visbook id.

So this method returns an array of SupabaseCabins, but these could really be entered into any database.

How about CabinSummary and getCabinSummaries? Other ideas welcome!

Reply:
That sounds great